_test/SyntaxTest.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace dokuwiki\plugin\bookcreator\test;
+
+use Doku_Handler;
+use DokuWikiTest;
+use syntax_plugin_bookcreator;
+
+/**
+ * Tests for the syntax component of the bookcreator plugin
+ *
+ * @group plugin_bookcreator
+ * @group plugins
+ */
+class SyntaxTest extends DokuWikiTest
+{
+
+    public function syntaxProvider() {
+        return [
+            'archive syntax with full normal details' => [ '~~ARCHIVEBOOK:5&date~~', [ 'archive', 5, 'date' ] ],
+            'archive syntax with full reversed details' => [ '~~ARCHIVEBOOK:title&7~~', [ 'archive', 7, 'title' ] ],
+            'archive syntax with count only' => [ '~~ARCHIVEBOOK:8~~', [ 'archive', 8, 'date' ] ],
+            'archive syntax with order pnly' => [ '~~ARCHIVEBOOK:title~~', [ 'archive', 10, 'title' ] ],
+            'archive syntax with defaults' => [ '~~ARCHIVEBOOK~~', [ 'archive', 10, '' ] ],
```

Comment:
I'm not sure that this is actually correct. Shouldn't the third element in the results array be `date`? This may be a bug in the code